### PR TITLE
CPP-1695: Serverless deploy prioritise env vars from secrets management service

### PR DESCRIPTION
**Pull request by joelcarr**
# Description

The CI environment was overwriting the values needed for production / other remote environments.

# Checklist:

- [x] My branch has been rebased onto the latest commit on main (don't merge main into your branch)
- [x] My commit messages are [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/), for example: `feat(circleci): add support for nightly workflows`, `fix: set Heroku app name for staging apps too`

### CODE REVIEW

**Review by ivomurrell — CHANGES_REQUESTED** [review]
**suggestion**: [looking](https://github.com/Financial-Times/dotcom-tool-kit/blob/dc016267972f89d0e38aa1979abd07693f831960/plugins/heroku/src/setConfigVars.ts#L29) at the Heroku plugin I think it would be better to ignore `process.env` entirely when deploying. Vault/Doppler is the source of truth for what is configured in Heroku/AWS and the whatever is set in CircleCI should be ignored.

File: plugins/serverless/src/tasks/deploy.ts [hunk 1]
```diff
@@ -36,8 +36,7 @@ export default class ServerlessDeploy extends Task<typeof ServerlessSchema> {
 
       const child = spawn('serverless', args, {
         env: {
-          ...dopplerEnv,
-          ...process.env
+          ...dopplerEnv
         }
```
Comment (ivomurrell) [inline plugins/serverless/src/tasks/deploy.ts:40]: **nitpick**: we don't need to spread the object anymore

```suggestion
        env: dopplerEnv
```

File: plugins/serverless/src/tasks/deploy.ts
Comment (ivomurrell) [inline plugins/serverless/src/tasks/deploy.ts:1]: **thought**: does the `useVault` option actually make any sense here? we don't seem to provide an analogous option in Heroku and and I don't know what the environment should look like if we were to skip Vault/Doppler...

**Review by ivomurrell — APPROVED** [review]
two non blocking comments but fine to merge

File: plugins/serverless/src/tasks/deploy.ts
Comment (joelcarr) [reply plugins/serverless/src/tasks/deploy.ts:1]: We want to get environment variables from the secrets management service to inject into the environment that `serverless.yml` will get the secrets from. Ideally we would use AWS Parameter Store instead meaning we will not need to fetch secrets from the secrets management service but we still need to figure out what that process will look like.

**Review by ivomurrell — APPROVED** [review]

**Pull request merged**